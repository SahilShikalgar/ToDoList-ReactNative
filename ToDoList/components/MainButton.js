import React from 'react'
import { TouchableOpacity, TouchableNativeFeedback, Platform, StyleSheet, View, Text } from 'react-native';
import Colors from '../constants/Colors';

export default function MainButton(props) {
    let ButtonComponent = TouchableOpacity;

    if (Platform.Version >= 21) {
        ButtonComponent = TouchableNativeFeedback;
    }

    return (
        <View style={styles.buttonContainer}>
            <ButtonComponent activeOpacity={0.6} onPress={props.onPress}>
                <View style={styles.button}>
                    <Text style={styles.buttonText}>{props.children}</Text>
                </View>
            </ButtonComponent>
        </View>
    );
}

const styles = StyleSheet.create({
    buttonContainer: {
        borderRadius: 25
    },
    button: {
        backgroundColor: Colors.primary,
        paddingVertical: 15,
        paddingHorizontal: 45,
        borderRadius: 25
    },
    buttonText: {
        color: 'white',
        fontFamily: 'lato',
        fontSize: 24,
        textAlign: 'center'
    }
});
