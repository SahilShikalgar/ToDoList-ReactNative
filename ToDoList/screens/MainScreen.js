import React, { useState } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import MainButton from '../components/MainButton';
import { FlatList } from 'react-native-gesture-handler';

export default function MainScreen(props) {
    const [toDoList, setToDoList] = useState([]);
    const title = props.navigation.getParam('title');
    const [itemAddedStatus, setItemAddedStatus] = useState(false);

    if (!itemAddedStatus) {
        setToDoList([...toDoList, title]);
        setItemAddedStatus(true);
        // use localStorage or Redux
    }

    const onPressHandler = () => {
        props.navigation.navigate('AddItem');
    }

    return (
        <View style={styles.screen}>
            <MainButton onPress={onPressHandler}>
                <Ionicons name="md-add" size={24} color="white" />&nbsp;
                ADD
            </MainButton>
        </View>
    )
}

const styles = StyleSheet.create({
    screen: {
        flex: 1,
        justifyContent: "center",
        flexDirection: 'row'
    }
});
