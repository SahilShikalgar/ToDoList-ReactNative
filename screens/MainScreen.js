import React, { useState } from 'react';
import { View, Text, StyleSheet, BackHandler } from 'react-native';
import MainButton from '../components/MainButton';
import { FlatList } from 'react-native-gesture-handler';
import { useSelector } from 'react-redux';

function Item({title}) {
    return (
        <View style={styles.listItem}>
            <Text style={styles.title}>{title}</Text>
        </View>
    );
}  


export default function MainScreen(props) {
    const [toDoList, setToDoList] = useState([]);
    const title = props.navigation.getParam('title');
    const [itemAddedStatus, setItemAddedStatus] = useState(false);

    const toDoListFromStore = [...useSelector(state => state.toDo.ToDoList)].reverse();

    if (!itemAddedStatus) {
        setToDoList([...toDoList, title]);
        setItemAddedStatus(true);
    }

    const onPressHandler = () => {
        props.navigation.navigate('AddItem');
    }

    return (
        <View style={styles.screen}>
            <MainButton onPress={onPressHandler} style={styles.button}>
                ADD
            </MainButton>
            <View style={styles.listContainer}>
                <FlatList
                    data={ toDoListFromStore }
                    renderItem={({ item }) => <Item title={item} />}
                    keyExtractor={item => Math.random(1,100) + item}
                    contentContainerStyle={styles.list}
                />
            </View>
            <MainButton
                onPress={onPressHandler} 
                style={{width: "50%"}} 
                onPress={() => { BackHandler.exitApp() }}
                style={styles.button}
            >
                EXIT
            </MainButton>
        </View>
    )
}

const styles = StyleSheet.create({
    screen: {
        flex: 1,
        justifyContent: "center",
        flexDirection: 'column',
        marginTop: "5%"
    },
    list: {
        flexGrow: 1,
        justifyContent: 'flex-start',
        marginTop: '5%'
    },
    listItem: {
        borderColor: '#ccc',
        borderWidth: 1,
        padding: 15,
        marginVertical: 10,
        backgroundColor: 'white',
        flexDirection: 'row',
        justifyContent: 'space-between',
        width: '100%'
    },
    listContainer: {
        flex: 1,
        width: '90%',
        alignSelf: "center",
        height: '70%',
        paddingVertical: 15
    },
    title: {
        fontFamily: 'lato'
    },
    button: {
        width: "50%",
        alignSelf: "center"
    }
});
