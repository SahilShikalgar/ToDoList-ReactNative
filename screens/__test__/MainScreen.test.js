import React from 'react';
import { shallow } from 'enzyme';
import { MainScreen, Item } from '../MainScreen';
import MainButton from '../../components/MainButton';
import { View } from 'react-native';

describe('MainScreen', () => {
    const navigation = {
        navigate: jest.fn(),
        getParam: jest.fn()
    }
    const toDo = ["Learn Mock Data"];
    const component = shallow (
        <MainScreen navigation={navigation} toDo={toDo} />
    );

    it('should render properly', () => {
        expect(component).toMatchSnapshot();
    });

    it('should navigate on add click', () => {
        component.find(MainButton).at(0).prop('onPress')();
        expect(navigation.navigate).toHaveBeenCalled();
    });

    it('should load mock Item component', () => {
        const itemComponent = shallow(<Item title={"List Item"} />);
        expect(itemComponent).toMatchSnapshot();
        const view = component.find(View);
        expect(view).toHaveLength(2)
    });
});
