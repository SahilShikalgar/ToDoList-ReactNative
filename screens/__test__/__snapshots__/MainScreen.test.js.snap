// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen should load mock Item component 1`] = `ShallowWrapper {}`;

exports[`MainScreen should render properly 1`] = `ShallowWrapper {}`;
